lang-ref/part1.md

```diff
@@ -0,0 +1,941 @@
+
+```
+function desc(Err err) returns string {
+    return err is XErr ? "X" : "Y";
+}
+```
+<br><br>
+
+## Panics
+
+In Ballerina, there is a distinction between normal and abnormal errors. Normal errors are the ones that you typically handle as part of the usual business logic and are under your control. But some errors are not under the control of the programmer. These are abnormal errors. Things such as bugs in library code or out of memory errors fall under this category.
+
+Normal errors are handled by returning values of the error type. Abnormal errors are handled using the panic statement.
+
+```
+function divide(int m, int n) returns int {
+    if n == 0 {
+        panic error("division by 0");
+    }
+    return m/n;
+}
+```
+ 
+A panic statement results in immediate program termination and has an associated error value.<br><br>
+
+## ``any`` type
+
+Ballerina also allows an ``any`` type, which means any value except an error. A variable of type any can be cast to a specific type using the ``‘< >’`` symbol enclosure.
+
+```
+any x = 1;
+
+int n = <int>x;
+```
+
+It can be converted to string.
+
+```
+string s = x.toString( );
+```
+
+You can also use the ``is`` operator to test its type.
+
+```
+float f = x is int|float ? <float>x : 0.0;
+```
+<br><br>
+
+## Ignoring return values and errors
+
+Ballerina does not allow silently ignoring return values of functions unless they return nil.
+
+```
+// Allowed only if return value is ()
+doX( );
+```
+
+To ignore the return value, you can assign it to `‘_’`, which acts as an implicitly declared variable of any type that cannot be referenced.
+
+```
+_ = getX( );
+```
+
+When a function returns an error type, it has to be handled. ``checkpanic`` keyword can be used otherwise but it panics on error rather than returning.
+
+```
+checkpanic tryX( );
+```
+<br><br>
+
+## Covariance
+
+Ballerina's typing system follows a structured type system based on the set of values that belongs to a type. For example, the integer type allows a set of values. Similarly, the any type also allows a set of values, including the subset of values that belong to the integer type.
+
+Therefore, you can have an array of integer type that is assigned to an array of any type.
+
+```
+int [ ] iv = [ 1, 2, 3];
+
+any [ ] av = iv;
+```
+  
+This is perfectly valid in Ballerina because the set of values allowed by an array of integer type is a subset of the values allowed by an array of any type. So this way, you can have some flexibility in allowing fixed inherent type substitution based on the set of values.
+
+Ballerina’s static type checking ensures that the result of a read operation on the array  **``av``** will be consistent with static type.  However, writing will result in an error because the Array iv has an inherent type set to integer . As a result, this code will lead to a runtime error since it is trying to mutate the Array to a type other than the inherent type.
```

Comment:
```suggestion
Ballerina’s static type checking ensures that the result of a read operation on the array  **``av``** will be consistent with static type. However, writing will result in an error because the Array iv has an inherent type set to integer. As a result, this code will lead to a runtime error since it is trying to mutate the Array to a type other than the inherent type.
```